app/mixins/confirm.js

```diff
@@ -0,0 +1,33 @@
+import Mixin from '@ember/object/mixin';
+import { isPresent } from '@ember/utils';
+import { get } from '@ember/object';
+
+export default Mixin.create({
+  modelName: null,
+  message: 'You will lose any unsaved information if you leave this page. Are you sure?',
+
+  actions: {
+    willTransition(transition) {
+      let modelName = get(this, 'modelName');
+      if (isPresent(modelName)) {
+        let model = get(this, `controller.${modelName}`);
+        if (get(model, 'isNew')) {
+          this._confirmTransition(transition, model);
+        }
+      } else {
+        this._confirmTransition(transition);
```

Comment:
Sorry I'm just now saying this, but it looks like this mixin has an edge case that's not covered here: https://coveralls.io/builds/14663763/source?filename=app%2Fmixins%2Fconfirm.js